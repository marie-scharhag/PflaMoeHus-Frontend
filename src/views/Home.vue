<template>
  <div class="home">
    <h1>Herzlich Wilkommen im PflaMoeHus</h1>
    <RoomList />
  </div>
</template>

<script>
import RoomList from "@/components/RoomList.vue";

export default {
  name: "Home",
  components: { RoomList },
};
</script>

<style scoped lang="scss">
h1{
  padding: 1.7em;
  text-align: center;
}
</style>


