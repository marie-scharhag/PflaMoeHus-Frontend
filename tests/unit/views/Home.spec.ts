import { shallowMount } from "@vue/test-utils"
import Home from "../../../src/views/Home.vue"

describe('Test for the Homepage',() => {

    const cmp = shallowMount(Home)

    test('If HTML structur is right', () => {
        expect(cmp.html()).toMatchSnapshot()
    })


    test('stubs a component, component RoomList is used in Home.vue + contained in html', () => {

        const RoomListMock = {	
            name: 'RoomList',
            template: 'fakeRoomList'
        }
        const wrapper = shallowMount(Home, {
            global: {
            stubs: { RoomList: RoomListMock }
            }
        }) 
        expect(wrapper.html()).toContain(`fakeRoomList`) 
        wrapper.findComponent(RoomListMock)
    })
})