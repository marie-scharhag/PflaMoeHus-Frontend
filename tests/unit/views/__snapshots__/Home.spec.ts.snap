// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the Homepage If HTML structur is right 1`] = `
<div class="home">
  <h1>Herzlich Wilkommen im PflaMoeHus</h1>
  <room-list-stub></room-list-stub>
</div>
`;
